app/models/book.rb

```diff
@@ -33,8 +33,9 @@ class Book < ActiveRecord::Base
   }
 
   def calculate_avg_score
+    size = ratings.size
     sum = ratings.inject(0.0) { |accum, rating| accum + rating.score }
-    sum / ratings.size
+    update_attributes(avg_score: sum / size)
```

Comment:
No, don't calculate the average in the application, use [average](http://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-average) to get this value